Mathlib/NumberTheory/LSeries/Deriv.lean

```diff
@@ -0,0 +1,165 @@
+private lemma LSeries.LSeriesSummable_logMul_and_hasDerivAt {f : ℕ → ℂ} {s : ℂ}
+    (h : abscissaOfAbsConv f < s.re) :
+    LSeriesSummable (logMul f) s ∧ HasDerivAt (LSeries f) (-LSeries (logMul f) s) s := by
+  -- The L-series of `f` is summable at some real `x < re s`.
+  obtain ⟨x, hxs, hf⟩ := LSeriesSummable_lt_re_of_abscissaOfAbsConv_lt_re h
+  obtain ⟨y, hxy, hys⟩ := exists_between hxs
+  -- We work in the right half-plane `y < re z`, for some `y` such that `x < y < re s`, on which
+  -- we have a uniform summable bound on `‖term f z ·‖`.
+  let S : Set ℂ := {z | y < z.re}
+  have h₀ : Summable (fun n ↦ ‖term f x n‖) := summable_norm_iff.mpr hf
+  have h₁ (n) : DifferentiableOn ℂ (term f · n) S :=
+    fun z _ ↦ (hasDerivAt_term f n _).differentiableAt.differentiableWithinAt
+  have h₂ : IsOpen S := isOpen_lt continuous_const continuous_re
+  have h₃ (n z) (hz : z ∈ S) : ‖term f z n‖ ≤ ‖term f x n‖ :=
+    norm_term_le_of_re_le_re f (by simpa using (hxy.trans hz).le) n
+  have H := hasSum_deriv_of_summable_norm h₀ h₁ h₂ h₃ hys
+  simp_rw [(hasDerivAt_term f _ _).deriv] at H
+  refine ⟨summable_neg_iff.mp H.summable, ?_⟩
+  have H' := differentiableOn_tsum_of_summable_norm h₀ h₁ h₂ h₃
+  simpa only [← H.tsum_eq, tsum_neg]
+    using (H'.differentiableAt <| IsOpen.mem_nhds h₂ hys).hasDerivAt
+
+/-- If `re s` is greater than the abscissa of absolute convergence of `f`, then the L-series
+of `f` is differentiable with derivative the negative of the L-series of the point-wise
+product of `log` with `f`. -/
+lemma LSeries.hasDerivAt {f : ℕ → ℂ} {s : ℂ} (h : abscissaOfAbsConv f < s.re) :
+    HasDerivAt (LSeries f) (- LSeries (logMul f) s) s :=
+  (LSeriesSummable_logMul_and_hasDerivAt h).2
+
+/-- If `re s` is greater than the abscissa of absolute convergence of `f`, then
+the derivative of this L-series at `s` is the negative of the L-series of `log * f`. -/
+protected
+lemma LSeries.deriv {f : ℕ → ℂ} {s : ℂ} (h : abscissaOfAbsConv f < s.re) :
+    deriv (LSeries f) s = - LSeries (logMul f) s :=
+  (hasDerivAt h).deriv
+
+/-- The derivative of the L-series of `f` agrees with the negative of the L-series of
+`log * f` on the right half-plane of absolute convergence. -/
+protected
+lemma LSeries.deriv_eqOn {f : ℕ → ℂ} :
+    {s | abscissaOfAbsConv f < s.re}.EqOn (deriv (LSeries f)) (- LSeries (logMul f)) :=
+  deriv_eqOn (isOpen_rightHalfPlane _) fun _ hs ↦ (hasDerivAt hs).hasDerivWithinAt
+
+/-- If the L-series of `f` is summable at `s` and `re s < re s'`, then the L-series of the
+point-wise product of `log` with `f` is summable at `s'`. -/
+lemma LSeriesSummable.logMul_of_lt_re {f : ℕ → ℂ} {s : ℂ} (h : abscissaOfAbsConv f < s.re) :
+    LSeriesSummable (logMul f) s :=
+  (LSeriesSummable_logMul_and_hasDerivAt h).1
+
+/-- The abscissa of absolute convergence of the point-wise product of `log` and `f`
+is the same as that of `f`. -/
+@[simp]
+lemma abscissaOfAbsConv_logMul {f : ℕ → ℂ} :
```

Comment:
I don't think anything here is dot-notatable, since `LSeries` is not Type-valued; but there might be an argument for putting stuff in the `LSeries` namespace anyway just for organisational purposes.

Reply:
I'll have a look and try to come up with something consistent. (Right now, I am travelling, so this will have to wait a couple of days.)

Reply:
The current naming scheme looks fine to me